src/_pytest/config/findpaths.py

```diff
@@ -93,104 +97,114 @@ def locate_config(
     ]
     args = [x for x in args if not str(x).startswith("-")]
     if not args:
-        args = [py.path.local()]
+        args = [Path.cwd()]
     for arg in args:
-        arg = py.path.local(arg)
-        for base in arg.parts(reverse=True):
+        argpath = absolutepath(arg)
+        for base in itertools.chain((argpath,), reversed(argpath.parents)):
             for config_name in config_names:
-                p = base.join(config_name)
-                if p.isfile():
+                p = base / config_name
+                if p.is_file():
                     ini_config = load_config_dict_from_file(p)
                     if ini_config is not None:
                         return base, p, ini_config
     return None, None, {}
 
 
-def get_common_ancestor(paths: Iterable[py.path.local]) -> py.path.local:
-    common_ancestor = None  # type: Optional[py.path.local]
+def get_common_ancestor(paths: Iterable[Path]) -> Path:
+    common_ancestor = None  # type: Optional[Path]
     for path in paths:
         if not path.exists():
             continue
         if common_ancestor is None:
             common_ancestor = path
         else:
-            if path.relto(common_ancestor) or path == common_ancestor:
+            if common_ancestor in path.parents or path == common_ancestor:
                 continue
-            elif common_ancestor.relto(path):
+            elif path in common_ancestor.parents:
                 common_ancestor = path
             else:
-                shared = path.common(common_ancestor)
+                shared = commonpath(path, common_ancestor)
                 if shared is not None:
                     common_ancestor = shared
     if common_ancestor is None:
-        common_ancestor = py.path.local()
-    elif common_ancestor.isfile():
-        common_ancestor = common_ancestor.dirpath()
+        common_ancestor = Path.cwd()
+    elif common_ancestor.is_file():
+        common_ancestor = common_ancestor.parent
     return common_ancestor
 
 
-def get_dirs_from_args(args: Iterable[str]) -> List[py.path.local]:
+def get_dirs_from_args(args: Iterable[str]) -> List[Path]:
     def is_option(x: str) -> bool:
         return x.startswith("-")
 
     def get_file_part_from_node_id(x: str) -> str:
         return x.split("::")[0]
 
-    def get_dir_from_path(path: py.path.local) -> py.path.local:
-        if path.isdir():
+    def get_dir_from_path(path: Path) -> Path:
+        if path.is_dir():
             return path
-        return py.path.local(path.dirname)
+        return path.parent
+
+    def safe_exists(path: Path) -> bool:
+        # On Python<=3.7, this can throw on paths that contain characters
```

Comment:
Perhaps we should use `sys.version_info` here as guard, so `safe_exists` can be dropped once only Python 3.8+ is supported?

```python
if sys.version_info[:2] <= (3, 7):
    def safe_exists(path: Path) -> bool:
        # On Python<=3.7, this can throw on paths that contain characters
        # unrepresentable at the OS level.
        try:
            return path.exists()
        except OSError:
            return False
else:
    def safe_exists(path: Path) -> bool:
        return path.exists()

```

Reply:
Good idea, will change it.

Reply:
BTW. The reason this can happen at all is pretty funky.

pytest determines the rootdir by looking at all non-option arguments and finding the common ancestor etc.

But because this code runs before plugins are loaded, any plugin argument not prefixed by a `-`, such as `test*bla` in `pytest --tx test*bla` (as opposed to if the `--tx=test*bla` form was used) gets interpreted as a possible path. If it is actually a valid path then you'd have one seriously confused user...

I looked into fixing it but seems not really possible.

Reply:
Yeah it is not simple due to how options are parsed. 😕